### PR TITLE
Feature/edit homepage review actions

### PR DESCRIPTION
**Pull request by rav-pradhan**
### What

This PR builds on and therefore includes the work in `feature/edit-homepage-add-edit-handlers` and `feature/editable-list-limit` too. Reviewing those PRs beforehand should mean this one will be shorter too if you wanted to review it like that. In any case, once approved, this will be merged into `feature/edit-homepage` which will then contain all of the work implemented for the new edit homepage form. That can then be subsequently PR'ed to be merged into Develop.

Those PRs include:
- Add/edit/cancel/delete handlers for the Headlines section of the edit homepage
- Incorporating a limit feature into the SimpleEditableList component

This PR furthermore includes:
- Extracted `DatasetReviewActions` logic into a more generic, `SaveAndReviewActions` component
- Additional unit tests for the Edit Homepage work
- Submit handlers logic
- Updates to the collections class for interfacing with Zebedee with JSON page content
- Update to CollectionDetails so that clicking on "Edit" or "Review" when the collection item is the homepage now routes you to the new Edit Homepage form rather than the workspace

### How to review

- Check code makes sense

You should be able to test this in the UI:
- Navigating to `/collections/{collectionID}/homepage` and then making your changes there
- Clicking Save and Preview should take you to the preview page and save your work too
- Clicking Save and Submit for Review should save your work and redirect you to the homepage
- Viewing a "Complete" page as the last author should still show Save and Submit for Review
- If you log in as another user, you should now be able to navigate to the homepage and see "Submit for Approval"
- All data should persist upon edits and subsequent saves

### Who can review

Anyone but me

### CODE REVIEW

File: src/app/utilities/api-clients/collections.js [hunk 1]
```diff
@@ -30,6 +30,27 @@ export default class collections {
         return http.put(`/zebedee/collection/${collectionID}`, body);
     }
 
+    static getContent(collectionID) {
+        return http.get(`/zebedee/data/${collectionID}?uri=/`);
```
Comment (jondewijones) [inline src/app/utilities/api-clients/collections.js:34]: With the param `?uri=/` isn't this only getting homepage content?

File: src/app/utilities/api-clients/collections.js [hunk 2]
```diff
@@ -30,6 +30,27 @@ export default class collections {
...
+    }
+
+    static getContentCollectionDetails(collectionID) {
+        return http.get(`/zebedee/collection/${collectionID}`);
+    }
+
+    static submitPageContentForReview(collectionID, contentURI) {
```
Comment (jondewijones) [inline src/app/utilities/api-clients/collections.js:41]: We already have methods called `setDatasetStatusToComplete`, `setDatasetStatusToReviewed`, `setDatasetVersionStatusToComplete` and `setDatasetVersionStatusToReviewed`. Could we be consistent? Perhaps zsetContentStatusToComplete` or change dataset methods names?

Comment (rav-pradhan) [reply src/app/utilities/api-clients/collections.js]: Sounds fair - have updated

File: src/app/utilities/api-clients/collections.js (hunk 1 above)
Comment (rav-pradhan) [reply src/app/utilities/api-clients/collections.js]: Yeah - realised this was a method no longer being used and had forgotten to remove it. Have removed it now

**Pull request merged**